Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -1131,6 +1137,41 @@ private void HandlePositionAssigned(OrderEvent fill)
             _algorithm.OnAssignmentOrderEvent(fill);
         }
 
+        /// <summary>
+        /// Option expiration event is received and new order events are generated
+        /// </summary>
+        private void HandleOptionPositionExpired(OptionPositionExpiredEventArgs e)
+        {
+            if (_algorithm.Securities.TryGetValue(e.Symbol, out var security))
+            {
+                Log.Trace("BrokerageTransactionHandler.HandleOptionPositionExpired(): clearing position for expired option holding: " +
+                    $"Symbol: {e.Symbol.Value}, Quantity: {security.Holdings.Quantity}");
+
+                var quantity = -security.Holdings.Quantity;
+
+                // generate new order and ticket for the expired option
+                var order = new OptionExerciseOrder(e.Symbol, quantity, DateTime.UtcNow)
```

Comment:
`DateTime.UtcNow` might no be friendly with backtesting, ~maybe the event has the time~ or maybe better use `CurrentTimeUtc` which the backtestingTH can override and return the algorithm UTC time?